core/primitives/src/shard_layout.rs

```diff
@@ -593,6 +593,8 @@ impl ShardLayout {
     /// Maps an account to the shard_id that it belongs to in this shard_layout
     /// For V0, maps according to hash of account id
     /// For V1 and V2, accounts are divided to ranges, each range of account is mapped to a shard.
+    /// Note: Calling function with receipt receiver_id to determine target shard is incorrect,
+    ///       `Receipt::receiver_shard` should be used instead.
```

Comment:
ah that is unfortunate

Reply:
Hmm wait, the global contract receipt doesn't have account id to begin with so it should be safe right? Does receiver_id return None or is not implemented for this receipt type?

Reply:
global contract distribution receipt always has `system` as `receiver_id`, so calling it `account_id_to_shard_id` would return incorrect shard_id

Reply:
ah yeah that's annoying. Perhaps we should assert `account_id_to_shard_id` isn't called with system? Or even better but not for this PR make account id an enum with different variant for system.

Reply:
asserting is a bit scary, it would be nice to solve that with better types